onnxruntime/test/util/default_providers.cc

```diff
@@ -47,25 +39,15 @@ std::unique_ptr<IExecutionProvider> DefaultMkldnnExecutionProvider(bool enable_a
 
 std::unique_ptr<IExecutionProvider> DefaultNupharExecutionProvider() {
 #ifdef USE_NUPHAR
-  OrtProviderFactoryInterface** f;
-  ORT_THROW_ON_ERROR(OrtCreateNupharExecutionProviderFactory(0, "", &f));
-  FACTORY_PTR_HOLDER;
-  OrtProvider* out;
-  ORT_THROW_ON_ERROR((*f)->CreateProvider(f, &out));
-  return std::unique_ptr<IExecutionProvider>((IExecutionProvider*)out);
+  return CreateExecutionProviderFactory_Nuphar(0, "")->CreateProvider();
 #else
   return nullptr;
 #endif
 }
 
 std::unique_ptr<IExecutionProvider> DefaultBrainSliceExecutionProvider() {
 #ifdef USE_BRAINSLICE
-  OrtProviderFactoryInterface** f;
-  ORT_THROW_ON_ERROR(OrtCreateBrainSliceExecutionProviderFactory(0, true, "testdata/firmwares/onnx_rnns/instructions.bin", "testdata/firmwares/onnx_rnns/data.bin", "testdata/firmwares/onnx_rnns/schema.bin", &f));
-  FACTORY_PTR_HOLDER;
-  OrtProvider* out;
-  ORT_THROW_ON_ERROR((*f)->CreateProvider(f, &out));
-  return std::unique_ptr<IExecutionProvider>((IExecutionProvider*)out);
+  return CreateExecutionProviderFactory_BrainSlice(0, true, "testdata/firmwares/onnx_rnns/instructions.bin", "testdata/firmwares/onnx_rnns/data.bin", "testdata/firmwares/onnx_rnns/schema.bin", &f));
```

Comment:
It's wrong?

Reply:
There's no exposed way to get a provider factory now, so I exposed a separate function that creates the providers directly on the inside. So that's why I removed the other code.

Reply:
You missed "->CreateProvider();"